### DISCUSSION TITLE
Mogrify convert GIF File to Transparency GIF File

### DISCUSSION THREAD

**Discussion by PanchoMariscal** [ImageMagick/ImageMagick#5853] [Digital Image Processing]
I am using mogrify from VBScript in a command line instruction, to made a GIF File to a Transparency GIF File,  
it convert about 4,000 images in one batch and run a lot of batches a day
eventually one of the processes create a temporary file under the TMP Directory called something like 
"magiz-Z......." with a 65Mb size and it contains mostly NULL Values, but my hard disk is becoming full with this  files
and the process stop executing.
I am attaching a ZIPPED sample file that is populating my  hard disk
How can I resolve this issue?

[magick-ZWfJabb6SfO-UfTvh_so0ZOo7yCj60gl.zip](https://github.com/ImageMagick/ImageMagick/files/10158152/magick-ZWfJabb6SfO-UfTvh_so0ZOo7yCj60gl.zip)

**Comment by fmw42** [comment]
Those files are created by Imagemagick and usually deleted automatically. But if a command is stopped prematurely, they get left behind. It is safe to delete them, for example at the end of the day or sooner. Or you can tell Imagemagick to use a different directory that may have more space allocated. See the Imagemagick Environment variables.

**Reply by PanchoMariscal** [reply]
Thanks fmw42, but this is a VBScrcipt that execute the following instrucion:
               set obj=CreateObject("ImageMagickObject.MagickImage.1")
               result=obj.Mogrify("-resize","561x316","-transparent","white","c:\mydirectory\*.gif")
in "c:\mydirectory" there are aprox 2500  GIF files ranking from 18Kb to 60Kb each one
that instruction is causing to fill the directory "C:\users\myuser\appdata\local\temp\" be filled with more than 2000 "mogrify-Z..." files
the C: files becomes  full for this cause

The fault becomes in one of the runs of this command

I dont have another disk with more space

Is there a way tho instruct ImageMagickObject to not produce this temporary files?


Thannks

**Comment by snibgo** [comment]
IM writes temporary files named "magick-*" when there isn't enough room in RAM for your pixels, so IM uses the disk instead. When IM completes, it removes those files.

But if IM is killed externally, it has no opportunity to delete temporary files. This may happen if the operating system discovers it has run out of resources, or some other reason.

I don't think the temporary file has useful diagnostic information. From your code and the files on your computer, perhaps you can identify the last file that was successfully processed, so the next file on the list caused the failure. Is something wrong with that file? Is it corrupt? Does it contain a stupid number of pixels?

Temporary files can be deleted whenever you like. Good system management practice is to remove them regularly: once per day, or week, or whatever.

**Reply by PanchoMariscal** [reply]
Thanks snibgo could you see my answer to  fmw42,
I observe each day my temporary files and there are not any temporary "mogrify-Z..." files so 
I cant clean in advance my local drive, 
I will try to do  your suggestion of identifying wich file is causing the problem, but I think I dont know in wich order is ImageMagick taking the files to process them

**Comment by fmw42** [comment]
You can run a cron job to automatically delete all files or just Imagemagick files in the directory either periodically or every night, for example, to clean that directory

**Reply by PanchoMariscal** [reply]
The problem fmw42 is that there are not temporary files before the command 
result=obj.Mogrify("-resize","561x316","-transparent","white","c:\mydirectory*.gif")

and there is not way a cron job can help me,  the problem is that in one single instruction is trying to convert more than 2500 files, this commando ussually works fine (about 90% times running) but there are some runs that cause internally this problem, with none temporary files  created before , and fill out the disk with this kind of files

**Reply by fmw42** [reply]
You may have one bad image that chokes mogrify.  Or perhaps you have a bad version of Imagemagick.  What is your version and platform/OS?  Have you tried smaller groups of images. That is more mogrify commands each on a smaller list of images?  Have you tried a script FOR loop over your images to see if convert/magick has the same issue as mogrify?

**Comment by snibgo** [comment]
> Is there a way to instruct ImageMagickObject to not produce this temporary files?

At the CLI, try: `-limit disk 0`. If IM needs disk, it will fail with an error message.

If that happens, as a fallback you could process each file one by one, ie not using `mogrify`.
